doc/rtd/explanation/kernel-cmdline.rst

```diff
@@ -3,20 +3,48 @@
 Kernel command line
 *******************
 
+Providing configuration data via the kernel command line is somewhat of a last
+resort, since this method supports only a subset of the
+:ref:`cloud config<user_data_cloud_config>` format, and many datasources
+do not support injecting kernel commandline arguments without modifying
+the bootloader.
+
+Despite the limitations of using the kernel commandline, cloud-init supports
+some use-cases.
+
+Note that this page describes kernel commandline behavior that applies
+to all clouds. To provide a local configuration with an image using kernel
+commandline, see :ref:`datasource NoCloud<datasource_nocloud>` which provides
+more configuration options.
+
+.. _kernel_datasource_override:
+
+Datasource discovery override
+=============================
+
+During boot, cloud-init must identify which datasource it is running on
+(OpenStack, AWS, Azure, GCP, etc). This discovery step can be optionally
+overriden by specifying the datasource name, such as:
+
+.. code-block:: text
+
+   root=/dev/sda ro ds=openstack
+
+Kernel cloud-config-url configuration
+=====================================
+
 In order to allow an ephemeral, or otherwise pristine image to receive some
-configuration, ``cloud-init`` will read a URL directed by the kernel command
+configuration, ``cloud-init`` can read a URL directed by the kernel command
 line and proceed as if its data had previously existed.
 
 This allows for configuring a metadata service, or some other data.
 
-.. note::
-   Usage of the kernel command line is somewhat of a last resort,
-   as it requires knowing in advance the correct command line or modifying
-   the boot loader to append data.
+This method is different from the standard way that most datasources receive
```

Comment:
Do we? I guess we could link to the `datasources.rst` file, is that what you mean?

Reply:
I can't find the place I thought it was. It might have been datasources.rst to be fair - or it might have been the instancedata.rst file in explanation. I'm not sure which is most appropriate so I'll leave that decision to you, but I do think we should link to somewhere where the standard way is outlined (however briefly).

Reply:
@s-makin - I'll just drop this sentence, since it's basically a duplication of the opening paragraph.